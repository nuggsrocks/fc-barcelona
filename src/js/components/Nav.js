import React from 'react'
import PropTypes from 'prop-types'

export const Nav = ({ routes, Link }) => {
  const logoSrc = 'icon.png'
  return (
    <nav>
      <img src={logoSrc} alt='FC Barcelona shield' />
      {
        routes.map(({ name, path }, index) =>
          <Link to={path} key={index} className='nav-link'>{name}</Link>
        )
      }
    </nav>
  )
}

Nav.propTypes = {
  routes: PropTypes.arrayOf(PropTypes.object),
  Link: PropTypes.object
}
