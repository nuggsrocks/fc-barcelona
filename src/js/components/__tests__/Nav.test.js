/* global React, render */
import { Link, BrowserRouter } from 'react-router-dom'
import { Nav } from '../Nav'

describe('<Nav/>', () => {
    it('should render logo in img element with links for each route', () => {
        const routes = [
            {name: 'foo', path: '/bar'}
        ]

        render(<BrowserRouter><Nav routes={routes} Link={Link} /></BrowserRouter>)

        expect(document.body).toMatchSnapshot()
    })
})