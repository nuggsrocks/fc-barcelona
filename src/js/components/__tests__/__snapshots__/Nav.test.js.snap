// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> should render logo in img element with links for each route 1`] = `
<body>
  <div>
    <nav>
      <img
        alt="FC Barcelona shield"
        src="icon.png"
      />
      <a
        class="nav-link"
        href="/bar"
      >
        foo
      </a>
    </nav>
  </div>
</body>
`;
